File: 351004/Hasiuk/Labs/src/main/java/com/example/Labs/entity/Story.java
package com.example.Labs.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "tbl_story")
@Getter @Setter
public class Story {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "editor_id", nullable = false)
    private Editor editor;
    @Column(nullable = false, length = 64)
    private String title;
    @Column(nullable = false, length = 2048)
    private String content;
    @Column(nullable = false)
    private LocalDateTime created;
    @Column(nullable = false)
    private LocalDateTime modified;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(name = "tbl_story_mark",
            joinColumns = @JoinColumn(name = "story_id"),
            inverseJoinColumns = @JoinColumn(name = "mark_id"))
    private Set<Mark> marks = new HashSet<>();
}

File: 351003/Soika/entities-app/publisher/src/main/kotlin/com/example/entitiesapp/model/Story.kt
package com.example.entitiesapp.model

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "tbl_story", schema = "distcomp")
data class Story(
    override var id: Long? = null,

    @Column(name = "writer_id", nullable = false)
    var writerId: Long,

    @Column(nullable = false, unique = true, length = 64)
    var title: String,

    @Column(nullable = false, length = 2048)
    var content: String,

    @Column(nullable = false)
    var created: LocalDateTime = LocalDateTime.now(),

    @Column(nullable = false)
    var modified: LocalDateTime = LocalDateTime.now(),

    @ManyToMany(fetch = FetchType.EAGER, cascade = [CascadeType.MERGE, CascadeType.PERSIST])
    @JoinTable(
        name = "tbl_story_mark",
        schema = "distcomp",
        joinColumns = [JoinColumn(name = "story_id")],
        inverseJoinColumns = [JoinColumn(name = "mark_id")]
    )
    var marks: MutableSet<Mark> = mutableSetOf()
) : BaseEntity(id)